### PR TITLE
Fix TODO: replace chmod

### PR DESCRIPTION
**Pull request by semelianova**

### CODE REVIEW

File: pkg/entities/runtask/actions.go [hunk 1]
```diff
@@ -226,9 +227,7 @@ func (action *uploadData) Apply() error {
 		return err
 	}
 
-	// TODO: use sftp.Chmod() instead
-	if err :=
-		runRemoteCommand(action.task.client, logger, "chmod", "+x", action.task.remotePath); err != nil {
+	if err := action.task.client.Chmod(action.task.remotePath, os.ModePerm); err != nil {
```
Comment (anmyachev) [inline pkg/entities/runtask/actions.go]: `os.ModePerm` is not equal to `+x`

**Comment by vnlitvinov** [comment]
@semelianova in addition to other commens please rebase upon `master` as I've merged your PR #21 so we now have CI

File: pkg/ssh/ssh.go [hunk 2]
```diff
@@ -245,6 +246,10 @@ func (client *zymeClient) PutFile(localPath, remotePath string, newlineConversio
 		}
 	}
 
+	if makeExecutable {
+		makeFileExecutable(*client, remotePath)
```
Comment (vnlitvinov) [inline pkg/ssh/ssh.go]: this should be checking for error, and _at least_ return it if it isn't `nil`

File: pkg/ssh/ssh.go [hunk 3]
```diff
@@ -353,3 +358,38 @@ func (client *zymeClient) ExecuteCommand(command string, getOutput bool) error {
 func (client *zymeClient) Split(path string) (dir, file string) {
 	return sftp.Split(path)
 }
+
+func makeFileExecutable(client zymeClient, path string) error {
+	sftpClient, err := sftp.NewClient(client.internalClient)
+	if err != nil {
+		log.WithFields(log.Fields{
+			"client": client.internalClient,
+		}).Errorf("ssh.makeFileExecutable: %s", err)
```
Comment (vnlitvinov) [inline pkg/ssh/ssh.go]:
```suggestion
		}).Errorf("ssh.makeFileExecutable: cannot create SSH client: %s", err)
```

File: pkg/ssh/ssh.go [hunk 4]
```diff
@@ -353,3 +358,38 @@ func (client *zymeClient) ExecuteCommand(command string, getOutput bool) error {
...
+		return err
+	}
+	defer sftpClient.Close()
+
+	fileInfo, err := sftpClient.Stat(path)
+	if err != nil {
+		log.WithFields(log.Fields{
+			"sftpClient": sftpClient,
+			"path":       path,
+		}).Errorf("ssh.makeFileExecutable: %s", err)
```
Comment (vnlitvinov) [inline pkg/ssh/ssh.go]:
```suggestion
		}).Errorf("ssh.makeFileExecutable: cannot stat '%s': %s", path, err)
```

File: pkg/ssh/ssh.go [hunk 5]
```diff
@@ -353,3 +358,38 @@ func (client *zymeClient) ExecuteCommand(command string, getOutput bool) error {
...
+		return err
+	}
+
+	fileModeX := fileInfo.Mode() | 0111
+	if err := sftpClient.Chmod(path, fileModeX); err != nil {
+		log.WithFields(log.Fields{
+			"sftpClient": sftpClient,
+			"path":       path,
+			"mode":       fileModeX,
+		}).Errorf("ssh.makeFileExecutable: %s", err)
```
Comment (vnlitvinov) [inline pkg/ssh/ssh.go]:
```suggestion
		}).Errorf("ssh.makeFileExecutable: cannot set executable bit on '%s': %s", path, err)
```

**Pull request merged**